### ISSUE TITLE
Field for user name not shown for public gallery

### ISSUE THREAD

**Issue by ariep** [ZenPhoto20/ZenPhoto20#428]
When the gallery is made "public" (instead of "private") and you are not logged in, you can see the gallery main page. This is as expected.

However, when you then want to log in (by clicking on the "log in" link at the bottom of the page), the login form does not show a user name field, only a password field. This seems weird; how is one supposed to log in in this scenario?

BTW, the gallery contains a single album, which is protected by a password (and no user). Perhaps this is relevant somehow.

**Comment by ariep** [comment]
I tried to create a new installation to test if this problem also occurs without the presence of any albums, but I could not get the new installation to work: when trying to run setup, I get "Fatal error: Call to undefined function zp_session_start() in /mnt/data/var/www/ciprpixra/zp-core/setup/index.php on line 37". If someone knows right away what is wrong there, I would be much obliged, but this is of course not directly related to the issue.

**Comment by sbillard** [comment]
There is an option on the security tab to force the presence of a User field for just this purpose. Normally the front-end logon is used for guest logins which don't necessarily have a user name associated. In that case the user field is not presented unless the "User name" option is checked. For private galleries there is no guest logon, so the field is always present.

I will look into the undefined function issue. Probably a problem of where the function is declared.

**Comment by sbillard** [comment]
I believe the zp_session_start issue is now fixed. Please verify.

**Comment by ariep** [comment]
> There is an option on the security tab to force the presence of a User field for just this purpose. Normally the front-end logon is used for guest logins which don't necessarily have a user name associated. In that case the user field is not presented unless the "User name" option is checked. For private galleries there is no guest logon, so the field is always present.

OK, I had not seen that option. Thanks.

I have a different problem now, though. Steps: log out, try to access the password-protected album, get a login screen, enter the correct album password (and no user). Result: the album page is shown, but instead of the images there is an error message:
```Fatal error: MySQLi Error: ( SELECT folder, `show` FROM `pixra_albums` WHERE id = ) failed. MySQLi returned the error You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1 in /mnt/data/var/www/pixra/zp-core/functions-common.php on line 55```

**Comment by ariep** [comment]
> I believe the zp_session_start issue is now fixed. Please verify.

Yes, this is indeed fixed. Thanks!

**Comment by ariep** [comment]
Unfortunately, my fresh installation is still not working. When I run setup, there appears the big list of setup checks. After fixing some of them (creating directories, fixing permissions, entering database details) I get a page saying "You must be logged in to run setup." with a login screen. Which is weird, because there aren't any users yet. Moreover, looking at the database it appears there are no tables created. I see no path forward from there.

**Comment by sbillard** [comment]
I am not able to reproduce the SQL error above. My best guess is that there is some issue with your database. Basically the error would occur if the **albumid** field of an image record is empty which should not be able to happen.

You can try a database refresh and see what happens. Also you can insert some debug code into class-search.php after line 1644 which reads ```$albumid = $row['albumid'];```

add ```if (empty($albumid)) debugLogVar($row);```

This would tell us what the image we are dealing with is.

**Comment by ariep** [comment]
> I am not able to reproduce the SQL error above. My best guess is that there is some issue with your database. Basically the error would occur if the albumid field of an image record is empty which should not be able to happen.

Yes, I had a look at the database contents, and this is exactly what has happened. Each image in the album occurs twice in the images table: once with a lower id and albumid set to NULL, and once with a higher id with the correct albumid. Moreover, a database refresh does not fix this.

Deleting the images with albumid NULL from the database manually fixed this problem.

**Comment by sbillard** [comment]
A database refresh should fix this, I will look into that. I have no idea how the situation could have existed in the first place. Maybe downstream from some other errors?

**Comment by ariep** [comment]
Yeah, my gallery seems to be in a screwed state somehow. I tried to add a new album (just creating a new directory with photo files in the "albums" directory), but I only get thumbnails consisting of a red cross, and errors that "the requested object was not found". It shouldn't be a permission error, because I access the image files on the webserver directly.

Because of all these weird problems, I tried  – unfortunately unsuccessfully – to start from scratch (see <https://github.com/ZenPhoto20/ZenPhoto20/issues/428#issuecomment-129213224>).
